File: interpreter.cc
#include "interpreter.h"
#include "controller.h"
#include "gamebook.h"
#include <sstream>

using namespace std;	
Interpreter::Interpreter(): openingBook("prepfile.txt"), gameBook("games.txt") {}

void Interpreter::initiate() {
	cout << "Welcome to this chess program!" << endl;
	cout << "Type \"help\" or \"help <cmd>\"" << endl;
	cout << "For general instructions or help" << endl;
	cout << "on a specific command." << endl;
	cout << "~> ";
	string input;
	while(getline(cin, input)) {
//		cout << input << endl; 		// ERASE THIS LATER
		stringstream ss(input);
		string command;
		ss >> command;
//		cout << command << endl; // ERASE THIS LATER TOO
		if(command == "q" || command == "quit") {
			cout << "Exiting." << endl;
			return;
		} else if(command == "help") {
			string arg;
			if(ss >> arg) {
				help(arg);	// This should be made into a default parameter and such to avoid: 
			} else {
				help("");	// this hacky thing. 
			}
		} else if(command == "openings") {
			fstream f;
			f.open(openingBook.c_str());
		   	fstream g;
			g.open(openingBook.c_str());
			cout << "Point A" << endl;
			Controller k(f, g, "opening");
			cout << "Point B" << endl;
			k.buildList();
			cout << "Point C" << endl;
			k.traverse();
			cout << "Point D" << endl;
			cout << "Done running the openings tool." << endl;
		} else if(command == "gamebook") {
			cout << "Running the gamebook tool" << endl;
			GameBook gb = GameBook(gameBook);
			gb.run();
			cout << "Done running the tool" << endl;
		} else if(command == "play") {
			cout << "Initiating a game." << endl;
			cout << "Done playing." << endl;
		} else {
			cout << "Command not recognized. Please type" << endl;
			cout << "\"help\" to see possible commands." << endl;
		}
		cout << "~> ";
	}
	cout << "Left while loop." << endl;
}

void Interpreter::help(string arg) {
	if(arg == "") {
		cout << "These are the current possible commands:" << endl;
		cout << "\"openings\": Use the opening book tool." << endl;
		cout << "\"gamebook\": Use the gamebook tool." << endl;
		cout << "\"play\": Play a game of chess." << endl;
	} // Have a bunch of else ifs to explain each possible command in detail.
}

File: src/com/skysketches/chess/Interpreter.java
/**
 * Project: Chess
 * Author: Ming-Cee Yee
 * Created: 2015-09-11
 * Description: Launches the Chess game
 */

package com.skysketches.chess;

import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.io.FileOutputStream;
import java.util.Scanner;
import java.util.StringTokenizer;

class Interpreter {
	
	private String openingBook; // TODO what are these		two variables that hold the names of the files that hold the openingbook and list of games. /what is an opening book
	private String gameBook; // name of the list of saved games TODO is there a way to make this into a reference instead of a string?

	public Interpreter() {
		this.openingBook = "prepfile.txt";
		this.gameBook = "games.txt";
	}

	/**
	 * init() initialises (launches) the game and provides main menu options
	 */
	public void init() {
		System.out.println("Welcome to this chess program!");
		try (Scanner scan = new Scanner(System.in)) {
		
			// accept a text command
			while (scan.hasNextLine()) {
				StringTokenizer line = new StringTokenizer(scan.nextLine());
				String command = line.nextToken();
				
				System.out.println("Type \"help\" for general instructions or " +
						"\"help <cmd>\" for help with a specific command.");
				System.out.println("~> ");
				
				// TODO catch invalid number of arguments
				if (command.equals("q") || command.equals("quit")) {
					System.out.println("Exiting game.");
					break;
				}
				else if (command.equals("help")) {
					if (line.hasMoreTokens()) {
						this.help(line.nextToken());
					}
					else {
						this.help();
					}
				}
				else if (command.equals("openings")) {
					System.out.println("Running the openings tool.");
					
					FileInputStream scanBookIn = new FileInputStream(openingBook);
					FileOutputStream scanBookOut = new FileOutputStream(openingBook);
					Controller controller = new Controller(scanBookIn, scanBookOut, "opening");
					
					controller.buildList();
					controller.traverse();
					System.out.println("Done running the openings tool.");
				}
				else if (command.equals("gamebook")) {
					System.out.println("Running the gamebook tool.");
					GameBook gb = new GameBook(gameBook);
					gb.run();
					System.out.println("Done running the gamebook tool.");
				}
				else if (command.equals("play")) {
					System.out.println("Initiating a game.");
					// TODO implement							// Yeah, this is something that has yet to be implemented, and is huge. 
					System.out.println("Done playing.");
				}
				else {
					System.out.println("Command not recognized. Please type " +
							"\"help\" to see possible commands.");
				}
			}
			
			System.out.println("Exited game.");
		}
		catch (FileNotFoundException e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
		}
	}
	
	/**
	 * help() lists all available commands
	 */
	public void help() {
		System.out.println("These are the current possible commands:");
		System.out.println("\"openings\": Use the opening book tool.");
		System.out.println("\"gamebook\": Use the gamebook tool.");
		System.out.println("\"play\": Play a game of chess.");
	}
	
	/**
	 * help(s) displays the description for the command s
	 */
	public void help(String s) {
		// TODO command descriptions
	}
}
